qiskit/circuit/quantumcircuit.py

```diff
@@ -806,10 +812,16 @@ def ancillas(self):
         """
         return self._ancillas
 
+    @deprecate_function('The QuantumCircuit.__add__() method is being deprecated.'
+                        'Use the compose() method which is more flexible w.r.t '
+                        'circuit register compatibility.')
```

Comment:
Not a big deal, but this will raise two errors on the user's first call to `__add__` (one about `__add__` being deprecated, one from `combine`). Could we warn only from `combine`/`extend` and have the message include that both `combine`/`extend` and their aliases as `+` and `+=` are being deprecated?

Reply:
Given how frequently `+` and `+=` are used, perhaps we can check if the user is depending (intentionally or implicitly) on the register extension behavior, and only warn only in that case?